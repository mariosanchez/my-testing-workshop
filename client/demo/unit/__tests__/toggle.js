import React from 'react'
import {mount, render} from 'enzyme'
import Toggle from '../toggle'

test('the component renders with defaults', () => {
  const wrapper = renderToggle()
  expect(wrapper).toMatchSnapshotWithGlamor()
})
test('onToggle function is call when button is clicked', () => {
  const onToggle = jest.fn()
  const wrapper = mountToggle(
    {
      onToggle,
    },
  )
  const button = wrapper.find(`[data-test="button"]`)
  expect(wrapper).toMatchSnapshotWithGlamor('1. Before toogle')
  button.simulate('click')
  expect(wrapper).toMatchSnapshotWithGlamor('2. After toogle')
  expect(onToggle).toHaveBeenCalledTimes(1)
  expect(onToggle).toHaveBeenCalledWith(true)
})

function renderToggle(props = {}) {
  const propsToUser = Object.assign(
    {
      onToggle() {},
      children: 'Child',
    },
    props,
  )
  return render(<Toggle {...propsToUser} />)
}

function mountToggle(props = {}) {
  const propsToUser = Object.assign(
    {
      onToggle() {},
      children: 'Child',
    },
    props,
  )
  return mount(<Toggle {...propsToUser} />)
}
