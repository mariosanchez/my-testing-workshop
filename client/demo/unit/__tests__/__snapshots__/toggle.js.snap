// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. Before toogle 1`] = `
.css-vp4bp7,
[data-css-vp4bp7] {
  text-align: center;
  display: inline-block;
  margin-bottom: 0px;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.4;
  padding: 6px 12px;
  cursor: pointer;
  border-radius: 4px;
  color: #fff;
  background-color: #337ab7;
  border-color: #285f8f;
}

<Toggle
  onToggle={[Function]}
>
  <glamorous(button)
    data-test="button"
    on={false}
    onClick={[Function]}
  >
    <button
      className="css-vp4bp7"
      data-test="button"
      onClick={[Function]}
    >
      Child
    </button>
  </glamorous(button)>
</Toggle>
`;

exports[`2. After toogle 1`] = `
.css-15i754,
[data-css-15i754] {
  text-align: center;
  display: inline-block;
  margin-bottom: 0px;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.4;
  padding: 6px 12px;
  cursor: pointer;
  border-radius: 4px;
  color: #fff;
  background-color: #22527b;
  border-color: #173853;
}

<Toggle
  onToggle={[Function]}
>
  <glamorous(button)
    data-test="button"
    on={true}
    onClick={[Function]}
  >
    <button
      className="css-15i754"
      data-test="button"
      onClick={[Function]}
    >
      Child
    </button>
  </glamorous(button)>
</Toggle>
`;

exports[`the component renders with defaults 1`] = `
.css-vp4bp7,
[data-css-vp4bp7] {
  text-align: center;
  display: inline-block;
  margin-bottom: 0px;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.4;
  padding: 6px 12px;
  cursor: pointer;
  border-radius: 4px;
  color: #fff;
  background-color: #337ab7;
  border-color: #285f8f;
}

<button
  class="css-vp4bp7"
  data-test="button"
>
  Child
</button>
`;
